tests/transactional/check_phub.pm

```diff
@@ -0,0 +1,36 @@
+# Copyright 2023 SUSE LLC
+# SPDX-License-Identifier: GPL-2.0-or-later
+#
+# Summary: Test packagehub repo on sle micro
+#
+# 1. Check if PackageHub extention is availble in sle micro
+# 2. Activate PackageHub extention
+# 3. Install a package from PackageHub repo
+#
+# Maintainer: qac <qa-c@suse.de>
+
+package check_phub;
+use base 'consoletest';
+use strict;
+use warnings;
+use testapi;
+use transactional;
+
+sub run {
+    select_console 'root-console';
+    my $out = script_output("transactional-update --quiet register --list-extensions|grep PackageHub", proceed_on_failure => 1);
+    if ($out ne "") {    #check if PackageHub extension is available
+        if ($out =~ /Activate with: transactional-update /p) {    #Check if it is activated
+            trup_call("${^POSTMATCH}");
+            process_reboot(trigger => 1);
+        }
+        trup_call("pkg install sshpass");
+        process_reboot(trigger => 1);
```

Comment:
Only minor thing if you don't mind, instead of `process_reboot`, can you use `check_reboot_changes;` ? This will ensure that there is a diff between the current snapshot and previous.
https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/lib/transactional.pm#L119

Reply:
Yes, I can change but one thing should be noted:
The transactional-update option: "--drop-if-no-change" is enabled by default for sle micro 5.4. 
Therefore, transactional-update in "trup_call" for 5.4 will diff snapshots and discard snapshot if there were no changes to the root file system. 

There is no need to check again with "check_reboot_changes" in this test.

Do you still want to change?

Reply:
The changes are expected. Either way, it would then fail to find the package

Reply:
Yes, in this case, we should get changes.

Reply:
Changed